RealmSwift/RealmConfiguration.swift

```diff
@@ -269,3 +264,39 @@ extension Realm.Configuration: CustomStringConvertible {
                     string: rlmConfiguration.description) ?? ""
     }
 }
+
+// MARK: Migration assistance
+
+public extension Realm.Configuration {
+    @available(*, unavailable, message: "Set `Realm.Configuration.kind` instead.")
```

Comment:
Wondering if, instead of making these unavailable, they should just be get-only properties.